diff --git a/ml/backend/ggml/ggml/src/ggml-cpu/cpu.go b/ml/backend/ggml/ggml/src/ggml-cpu/cpu.go
index 895b7f6..08cb989 100644
--- a/ml/backend/ggml/ggml/src/ggml-cpu/cpu.go
+++ b/ml/backend/ggml/ggml/src/ggml-cpu/cpu.go
@@ -1,7 +1,7 @@
 package cpu
 
-// #cgo CFLAGS: -O3 -Wno-implicit-function-declaration
-// #cgo CXXFLAGS: -std=c++17
+// #cgo CFLAGS: -O3 -Wno-implicit-function-declaration -march=native
+// #cgo CXXFLAGS: -std=c++17 -march=native
 // #cgo CPPFLAGS: -I${SRCDIR}/amx -I${SRCDIR}/llamafile -I${SRCDIR}/.. -I${SRCDIR}/../../include
 // #cgo CPPFLAGS: -DNDEBUG -DGGML_USE_LLAMAFILE
 // #cgo linux CPPFLAGS: -D_GNU_SOURCE
diff --git a/ml/backend/ggml/ggml/src/ggml.go b/ml/backend/ggml/ggml/src/ggml.go
index 91f1f1a..47c850e 100644
--- a/ml/backend/ggml/ggml/src/ggml.go
+++ b/ml/backend/ggml/ggml/src/ggml.go
@@ -1,6 +1,7 @@
 package ggml
 
-// #cgo CXXFLAGS: -std=c++17
+// #cgo CFLAGS: -march=native
+// #cgo CXXFLAGS: -std=c++17 -march=native
 // #cgo CPPFLAGS: -DNDEBUG -DGGML_USE_CPU
 // #cgo CPPFLAGS: -I${SRCDIR}/../include -I${SRCDIR}/ggml-cpu
 // #cgo windows CFLAGS: -Wno-dll-attribute-on-redeclaration
